File: 3/3.rs
use std::collections::HashSet;

fn primes(x: u64) -> HashSet<u64> {
    let root = (x as f64).sqrt() as u64 + 1;
    let mut results = HashSet::new();
    let mut i = 2u64;
    let mut n = x;
    while i < root && n > 1 {
        if n % i == 0 {
            results.insert(i);
            n = n / i;
        } else {
            i += 1;
        }
    }
    return results;
}

fn main() {
    let x = 600851475143;
    println!("{:#?}", primes(x).iter().max());
}

File: 3/3.go
package main

import (
	"fmt"
)

func getPrimes(x int) []int {
	var primes []int
	for i := 3; i * i < x; i += 2 {
		for x % i == 0 {
			primes = append(primes, i)
			x /= i
		}
	}
	primes = append(primes, x)
	return primes
}

func main() {
	fmt.Println(getPrimes(600851475143))
}
